sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConstants.scala

```diff
@@ -18,6 +18,7 @@ private object CosmosConstants {
   val maxRetryCountForTransientFailures = 100
   val defaultDirectRequestTimeoutInSeconds = 10L
   val feedRangesCacheIntervalInMinutes = 1
+  val defaultIoThreadCountFactorPerCore = 4
```

Comment:
Do we have tests about before vs after?

Reply:
No - we don't have a repro in-house where we clearly know IO-ThreadCount was the culprit.